controllers/core/node_controller.go

```diff
@@ -67,7 +69,9 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 	if err := r.Client.Get(ctx, req.NamespacedName, node); err != nil {
 		if errors.IsNotFound(err) {
-			_, found := r.Manager.GetNode(req.Name)
+			cachedNode, found := r.Manager.GetNode(req.Name)
+			// delete the not found node instance id from node event cache for housekeeping
+			r.deleteNodeFromNodeEventCache(cachedNode.GetNodeInstanceID())
```

Comment:
Don't we need error checks here? I assume we are getting from the node manager but what if node manager returns `nil` ?

Reply:
This is deleting the instance id from event cache key set. If that ops fails, we log the failure. But I think we can live with it and don't have to retry or bubble up the failure. The cache can evict old entries in the worst case.